### DISCUSSION TITLE
Help calculate tcp retransmission rate from json result, end. sum.sent

### DISCUSSION THREAD

**Discussion by megaxm** [esnet/iperf#1376] [Q&A] [answered]
Hi.
Could you please explain if it is possible to calculate the tcp retransmission rate. 
Given the following json result, end. sum.sent
`"sum_sent":     {
                        "start":        0,
                        "end":  60.000091,
                        "seconds":      60.000091,
                        "bytes":        2942828544,
                        "bits_per_second":      392376544.0955748,
                        "retransmits":  291,
                        "sender":       true
                }`
My understanding is that the field "retransmits" is in segments, so I would calculate the retransmission rate using
retransmits / total_segments_sent, but I only have the bytes_sent field, is there anyway to convert the number on the field bytes_sent into segments, or the filed retransmits into bytes for that matter, so that I can apply the same unit to get the rate?
Thanks so much in advance.

**Comment by davidBar-On** [comment]
Hi, pack length is fixed size and is set by the `-l` parameter (default is usually 128KB = 131,072 bytes).  Therefore dividing bytes by the packet size should work.  E.g, in the example data you gave and assuming the default packet length was used, 22,452 packets were sent.

**Reply by megaxm** [reply]
Thanks so much.

**Reply by megaxm** [reply]
Sorry David, one more thing.
I have the "blksize":      131072, in the result as you pointed out, but also the "tcp_mss" is set to 1514,
Should I still use blksize or tcp_mss? to get the the counts of packets Eg. packets=bytes/blksize or packets=bytes/tcp_mss.
then rate=retransmits/packets ?
Could you please let me know.
Thanks.

**Comment by davidBar-On** [comment, marked as answer]
Hi, the "tcp_mss" is the [socket's TCP_MAXSEG](https://www.masterraghu.com/subjects/np/introduction/unix_network_programming_v1.3/ch07lev1sec9.html) value, which is the segment size used by the TCP/IP stack for sending the data to the other side.  However, it is not related to the packet size used by iperf3.

To make sure, you can use the `-k` option to set the number of packets transmitted.  E.g. set `-k 10`, and if the packets=bytes/blksize is correct test results should show `"bytes": 1310720`.

**Reply by megaxm** [reply]
Excellent thanks again for your time.
